bin/packages/lint-staged-typecheck.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * External dependencies
+ */
+const execa = require( 'execa' );
+
+/* eslint-disable no-console */
+try {
+	execa.sync( 'npm', [ 'run', 'build:package-types' ] );
```

Comment:
It might help to include a comment somewhere in here explaining why this file exists, vs. calling the NPM script directly from the pre-commit task. Otherwise, I fear it may not be obvious to a future maintainer that it need to continue to exist. Basically, what you had discovered at https://github.com/WordPress/gutenberg/pull/18942/files#r397735268 .